packages/office-ui-fabric-react/src/components/ScreenReaderAlert/ScreenReaderAlert.tsx

```diff
@@ -0,0 +1,163 @@
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IScreenReaderAlertProps): void {
+    if (this._clearTextTimeout) {
+      // Clear the timeout of removing text to avoid the new text being removed.
+      window.clearTimeout(this._clearTextTimeout);
+      this._clearTextTimeout = undefined;
+    }
+
+    this.setState({
+      alertText: this._getTextFromProps(React.Children.toArray(nextProps.children))
+    });
+  }
+
+  public shouldComponentUpdate(nextProps: IScreenReaderAlertProps, nextState: IScreenReaderAlertState): boolean {
+    // When indicator is not changed, do NOT need to render the string if it's same with the previous non-empty string.
+    const isValidStringChanged: boolean = nextState.alertText !== this._previousValidString;
+
+    return (this.props.indicator !== nextProps.indicator || isValidStringChanged);
+  }
+
+  public componentDidUpdate(nextProps: IScreenReaderAlertProps): void {
+    this._clearTextByTimeout();
+    this._updateValidString();
+  }
+
+  public componentDidMount(): void {
+    this._clearTextByTimeout();
+    this._updateValidString();
+  }
+
+  public render(): JSX.Element {
+    return (
+      <div className={ styles.screenReaderAlert } key={ this._renderIndex++ }>
+        {
+          this.props.readingMode !== ReadingMode.DoNotRead && this.state.alertText && (
+            <p
+              role={ this._role }
+              aria-live={ this._ariaLive }
+              aria-atomic={ true }>
+              { this.state.alertText }
+            </p>
+          )
+        }
+      </div>
+    );
+  }
+
+  private get _role(): string {
+    return this.props.readingMode === ReadingMode.ReadImmediately ? 'alert' : undefined;
+  }
+
+  private get _ariaLive(): string {
+    switch (this.props.readingMode) {
+      case ReadingMode.DoNotRead:
+        return 'off';
+      case ReadingMode.ReadAfterOtherContent:
+        return 'polite';
+      case ReadingMode.ReadImmediately:
+        return 'assertive';
+      default:
+        return undefined;
+    }
+  }
+
+  private _getTextFromProps(root: React.ReactChild | React.ReactChild[]): string {
+    let text: string = '';
+    if (typeof root === 'string' || typeof root === 'number') {
+      text += root;
+    } else if (Array.isArray(root)) {
+      root.forEach((child: React.ReactChild) => text += this._getTextFromProps(child));
+    }
+
+    return text;
+  }
+
+  private _updateValidString(): void {
+    this._previousValidString = this.state.alertText || this._previousValidString;
+  }
+
+  // Set a timeout to remove the text to avoid the text persisting in the DOM which can be confusing.
+  // It will be cleared if props updating is occured before the text being cleared.
+  private _clearTextByTimeout(): void {
+    if (this.state.alertText) {
+      this._clearTextTimeout = window.setTimeout(() => {
+        this.setState({
+          alertText: ''
+        });
+        this._clearTextTimeout = undefined;
```

Comment:
I think this is unanticipated behavior and makes the use case really unclear. Here's how I understand this:

1. The text is rendered.
2. 1 second later, the text is removed.

Is that really expected? When will this be read? If it is removed before it gets to reading it, will it still be read? It seems like this could cause problems, especially for the 'polite' reading mode?